test/test_tensordict.py

```diff
@@ -2935,6 +2935,52 @@ def test_unflatten_keys_collision(separator):
     assert assert_allclose_td(td.unflatten_keys(separator), ref)
 
 
+def test_split():
+    td = TensorDict({"a": torch.zeros(10)}, [10])
+
+    tds = td.split(5, 0)
+    assert len(tds) == 2
+    assert tds[0].shape == torch.Size([5])
+    assert tds[1].shape == torch.Size([5])
+
+    tds = td.split([5, 2, 3], 0)
+    assert len(tds) == 3
+    assert tds[0].shape == torch.Size([5])
+    assert tds[1].shape == torch.Size([2])
+    assert tds[2].shape == torch.Size([3])
+
+    td = TensorDict(
+        source={"a": torch.zeros(5, 6, 5), "b": torch.zeros(5, 6, 10, 1)},
```

Comment:
I guess that if you swap the values of a and b, since you take the shape of the first tensor up to the last-but-one dimension, you'll end up with a batch size that is not accurate no?

Reply:
I agree, I'll take the shape of the first tensor up to the batch-size length to ensure the batch size reflects the first few common dimensions.

Reply:
Beware of the fact that the batch size is not necessarily the shape up to the last common dimension. It can be anything in between 0 and that.